### PR TITLE
combined arts swap

### PR DESCRIPTION
**Pull request by SoundSSGood**
swap

### CODE REVIEW

File: lib/CArtHandler.cpp [hunk 1]
```diff
@@ -1023,43 +1023,44 @@ bool CArtifactInstance::isPart(const CArtifactInstance *supposedPart) const
...
+	fittingSet.artifactsWorn = artSet->artifactsWorn;
+	auto removedArt = fittingSet.getArt(slot);
+	if(assumeDestRemoved && removedArt)
 	{
-		for(auto art = constituentsToBePlaced.begin(); art != constituentsToBePlaced.end(); art++)
+		if(removedArt->canBeDisassembled())
 		{
-			// pos == slot because we can remove already worn artifact only from that slot. That is our main destination
-			if(art->art->canBePutAt(artSet, pos, pos == slot))
+			auto contitutient = dynamic_cast<CCombinedArtifactInstance*>(removedArt);
```
Comment (IvanSavenko) [inline lib/CArtHandler.cpp:1040]: `contitutient`? Typo?
And from context, `combinedArtToRemove` is probably a better name

**Review by IvanSavenko — APPROVED** [review]
Looks OK code-wise.

To clarify, this change allows swapping two combined artifacts directly, which was not allowed before. Right?

File: lib/CArtHandler.cpp (hunk 1 above)
Comment (SoundSSGood) [reply lib/CArtHandler.cpp:1040]: > To clarify, this change allows swapping two combined artifacts directly, which was not allowed before. Right?
Yeah.
contitutient and combined - both of them occur in code. But "combined" better, I think. Will fix.

**Comment by nullkiller** [comment]
Please rebase on top of develop (or merge develop) so that build artifacts contain your other change

**Comment by nullkiller** [comment]
Tested, found no issues. So we still can not swap angelic alliance and armor of damned but swapping two armors works.

**Pull request merged**